contracts/modules/legacy/LegacyPooledStaking.sol

```diff
@@ -1393,10 +1387,15 @@ contract LegacyPooledStaking is IPooledStaking, MasterAwareV2 {
     uint nxmBalanceAfter = nxm.balanceOf(address(this));
     uint nxmToBeUnlocked = deposit - (nxmBalanceBefore - nxmBalanceAfter);
 
+    // Set deposit to zero to avoid re-entrancy
+    stakers[stakerAddress].deposit = 0;
+
     // Send unlocked NXM back
     nxm.transfer(stakerAddress, nxmToBeUnlocked);
   }
 
+  // TODO review if we want this functionality
+  // We might want to allow users to lock their deposit in multiple tranches
```

Comment:
Agreed currently this functionality is limited so not that useful, but given timeline it's a decision whether to have it now or just let them pull out and deposit on their own